atlasdb-timelock-server/src/integTest/java/com/palantir/atlasdb/timelock/TimeLockServerIntegrationTest.java

```diff
@@ -153,13 +152,38 @@ public void timestampServiceShouldThrowIfQueryingNonexistentClient() {
     }
 
     @Test
+    public void timestampMigrationServiceShouldThrowIfQueryingNonexistentClient() {
+        TimestampMigrationService nonexistent = getTimestampMigrationService(NONEXISTENT_CLIENT);
+        assertThatThrownBy(() -> nonexistent.fastForwardTimestamp(ONE_MILLION))
+                .hasMessageContaining(NOT_FOUND_CODE);
+    }
+
+    @Test
     public void timestampServiceShouldThrowIfQueryingInvalidClient() {
         TimestampService nonexistent = getTimestampService(INVALID_CLIENT);
         assertThatThrownBy(nonexistent::getFreshTimestamp)
                 .hasMessageContaining(NOT_FOUND_CODE);
     }
 
     @Test
+    public void timestampMigrationServiceShouldThrowIfQueryingInvalidClient() {
+        TimestampMigrationService nonexistent = getTimestampMigrationService(INVALID_CLIENT);
+        assertThatThrownBy(() -> nonexistent.fastForwardTimestamp(ONE_MILLION))
+                .hasMessageContaining(NOT_FOUND_CODE);
+    }
+
+    @Test
+    public void timestampMigrationShouldFastForwardForClientNamespaces() {
+        TimestampService timestampServiceForClient1 = getTimestampService(CLIENT_1);
+        TimestampMigrationService timestampMigrationServiceForClient2 = getTimestampMigrationService(CLIENT_2);
+
+        long firstTimestampForClient1 = timestampServiceForClient1.getFreshTimestamp();
+        timestampMigrationServiceForClient2.fastForwardTimestamp(ONE_MILLION);
+        long secondTimestampForClient1 = timestampServiceForClient1.getFreshTimestamp();
+        assertThat(secondTimestampForClient1).isGreaterThan(firstTimestampForClient1).isLessThan(ONE_MILLION);
```

Comment:
Observation: we use `both` someplace above, and this construction here. Either's fine by me.